test/Microsoft.NET.TestFramework/Utilities/FileThumbPrint.cs

```diff
@@ -61,7 +61,14 @@ public bool Equals(FileThumbPrint? other)
                 string.Equals(Hash, other.Hash, StringComparison.Ordinal);
         }
 
-        public override int GetHashCode() => LastWriteTimeUtc.GetHashCode();
+        public override int GetHashCode()
+        {
+#if NET
+            return HashCode.Combine(Path, LastWriteTimeUtc, Hash);
+#else
+            return Path.GetHashCode() ^ LastWriteTimeUtc.GetHashCode() ^ Hash.GetHashCode();
+#endif
+        }
```

Comment:
Is this simply a bug in the test infrastructure that we are fixing?

Also, VS will generate the GetHashCode for you in FullFramework. It generates something more "robust" like the "sample" below. We should probably just use VS to generate it.

```
        var hashCode = 1447485498;
        hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Path);
        hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(LastWriteTimeUtc);
        hashCode = hashCode * -1521134295 + Hash.GetHashCode();
```

Reply:
> Is this simply a bug in the test infrastructure that we are fixing?

We are enabling the xxx.runtimeconfig.json file to be generated for blazor and it has consequences. 

I was not sure what's the root cause of the difference. Now I know.

[Log](https://helixr1107v0xdeko0k025g8.blob.core.windows.net/dotnet-sdk-refs-pull-48916-merge-bae5abb28a014582ba/Microsoft.NET.Sdk.BlazorWebAssembly.Tests.dll.2/3/console.035d3b44.log?helixlogtype=result)
```
5/14/2025 4:46:58 PM should match 5/14/2025 4:46:52 PM for /private/tmp/helix/working/BAA1098D/t/dotnetSdkTests.rrWc2GDW/Build_IsIncre---61DA2120/blazorwasm/bin/Debug/net10.0/blazorwasm.runtimeconfig.json
```

Reply:
I filled https://github.com/dotnet/runtime/issues/115646 for the broken incrementalism